### DISCUSSION TITLE
Are there any use cases built around Targeted Sentiment analysis/ Entity Level Sentiment Analysis in Spacy

### DISCUSSION THREAD

**Discussion by thalishsajeed** [explosion/spaCy#8931] [Help: Coding & Implementations] [answered]
traditional NER models are able to identify all the entities in a given span of text. 

`I would love to meet John <PERSON> but I am not too keen on meeting Susan <PERSON>`

 I am looking to build a workflow that goes beyond this into [Entity Level Sentiment Analysis](https://cloud.google.com/natural-language/docs/analyzing-entity-sentiment)

I have seen it being called by other names such as - [Targeted Sentiment Analysis](https://towardsdatascience.com/targeted-sentiment-analysis-vs-traditional-sentiment-analysis-4d9f2c12a476)  

The objective would be to assign a sentiment score to each detected entity. With the new annotation interface for span categoriser in prodigy , I think I would be able to start a labelling exercise , However I wanted to understand if there's any way to use spancat or any other spacy component to perform the task I am attempting (or any other helpful tips on approaching this problem)

**Comment by thomashacker** [comment, marked as answer]
I'm currently working on a similar project where I determine effects of supplements on health conditions. 

First, all conditions are identified by an NER, afterwards a representative statement is extracted for every entity.
These extracted statements are then classified by a textcat whether the effect on the condition is positive or negative.

For your example you could do:

 I would love to meet `<PERSON>` | Positive Sentiment
 I am not too keen on meeting `<PERSON>` | Negative Sentiment

**Reply by thalishsajeed** [reply]
Hmm, Looking at the dataset, it seems to be tricky to identify sentence boundaries that would simplify the problem. Thanks for the tip. Will see if there's someway I can reduce the task to that of a sentence classification.

**Reply by thomashacker** [reply]
Clause Segmentation works great for my use case. 
Could you share some examples of your dataset and which sentiment you'd expect for every entity?

**Reply by thalishsajeed** [reply]
eg 1 : I am currently living in india (COUNTRY|NEUTRAL) and would like to learn more about germany (COUNTRY|POSITIVE)

so this is a somewhat straightforward example where I can use the previous tokens to predict sentiment for the following entity. 


eg2: Working in India (COUNTRY|NEGATIVE) right now and I am not too happy with it would like to explore GERMANY (COUNTRY|POSITIVE). 

the sentiment is held in the forward tokens. 

I think for starters I just would like NEGATIVE, POSITIVE and NEUTRAL sentiment.

**Reply by thomashacker** [reply]
That's a great example. I would suggest to deconstruct the sentence into three clauses like this:

1. Working in `<Country>` right now (NEUTRAL)
2. I am not too happy with it (NEGATIVE)
3. Would like to explore `<Country>` (POSITIVE)

Ideally, we also want to classify statements that don't contain any entities, so that we have more data to work with afterward.
The real challenge, as you said, is the connection between 1. and 2. With coreference resolution, you could try to resolve the "it" -> to -> "working in India" which would result in: I am not too happy with working in India. It would be interesting to see if this works 😄 

1. Working in `<Country>` right now (NEUTRAL)
2. I am not too happy with working in `<Country>` (NEGATIVE)
3. Would like to explore `<Country>` (POSITIVE)

You can also have a look at the [Berkeley Neural Parser](https://parser.kitaev.io/)  which provides the information we need for clause segmentation:

![image](https://user-images.githubusercontent.com/43848523/129143634-ab317d46-d383-4d12-92cb-c357433fae23.png)

**Reply by thalishsajeed** [reply]
Ah, yes co-reference might do the trick. Let me take a look at that as well.
